lib/gui/etcher.js

```diff
@@ -81,4 +82,32 @@ electron.app.on('ready', () => {
   })
 
   mainWindow.loadURL(`file://${path.join(__dirname, 'app', 'index.html')}`)
+}
+
+electron.app.on('window-all-closed', electron.app.quit)
+
+// Sending a `SIGINT` (e.g: Ctrl-C) to an Electron app that registers
+// a `beforeunload` window event handler results in a disconnected white
+// browser window in GNU/Linux and macOS.
+// The `before-quit` Electron event is triggered in `SIGINT`, so we can
+// make use of it to ensure the browser window is completely destroyed.
+// See https://github.com/electron/electron/issues/5273
+electron.app.on('before-quit', () => {
+  process.exit(EXIT_CODES.SUCCESS)
 })
+
+settings.load().then((localSettings) => {
+  Object.assign(config, localSettings)
+}).catch((error) => {
+  // TODO: What do if loading the config fails?
+  console.error('Error loading settings:')
+  console.error(error)
```

Comment:
We should probably `logDebug` and/or `logException` this as well, although the `console.error`s take care of the former.

Reply:
@Shou I don't think we can use analytics here, as this is the main entry point (and the analytics not set up yet), as well as there being no point to opt out of them before this runs

Reply:
Fair, let's just log them then.